internal/cloud/azure/application_identity.go

```diff
@@ -79,12 +80,13 @@ func CreateApplicationIdentity(ctx context.Context, config *ApplicationIdentityC
 		return spErr
 	}
 	config.ServicePrincipalID = *sp.ID
+	config.ServicePrincipalClientID = *sp.AppId
 	config.ID = *sp.ID
 
 	// fetch the service principal to make sure it exists
-	_, _, spCheckErr := spClient.Get(ctx, config.ServicePrincipalID, odata.Query{})
+	_, _, spCheckErr := spClient.Get(ctx, config.ID, odata.Query{})
 	if spCheckErr != nil {
-		return fmt.Errorf("error retrieving Service Principal with ID %v: %w", config.ServicePrincipalID, spCheckErr)
+		return fmt.Errorf("error retrieving Service Principal with ID %v: %w", config.ServicePrincipalClientID, spCheckErr)
```

Comment:
Why are we swapping the types here? We check if it exists using the `config.ID` then if it fails in the error message we're outputing a different value (`config.ServicePrincipalClientId`)

Reply:
I recall this was a change made when still trying to discover what the actual values contained. Next commit will change this value to `config.ID`